oauth2_lib/fastapi.py

```diff
@@ -10,20 +10,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```

Comment:
Haven't read everything in detail yet, but great work so far!

The problem (created largely by us) is that this file is a bit of a kitchen sink, almost everything authentication/authorization related is in here. This makes it difficult to review your work.

Both for that reason and for keeping this library maintainable, I think we should split things off to separate modules. Which we'd probably want to do on the 'old' code in a separate MR, merge that, and rebase your work on top of it.

If you agree, the next step would be to consider which modules to separate this into. Probably at least separate modules for: oidc_user, oidc_config, authn, authz.

Reply:
I'll try to draw up a PR for this later today

Reply:
I agree. That would be cool!
Let me know when you are done with that PR.